cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/MessagingMetricsCollectionService.java

```diff
@@ -176,5 +190,9 @@ private RetryStrategy getRetryStrategy() {
       // If failure happen during shutdown, use a retry strategy that only retry fixed number of times
       return RetryStrategies.timeLimit(5, TimeUnit.SECONDS, RetryStrategies.fixDelay(200, TimeUnit.MILLISECONDS));
     }
+
+    void setMetricsTags(Map<String, String> metricsTags) {
```

Comment:
Shouldn't need this method. You can have the context provided to the `addPayload` method (`addPayload(Map<String, String> context, byte[] payload)`)